nvdaHelper/local/UIAEventLimiter/rateLimitedEventHandler.cpp

```diff
@@ -0,0 +1,227 @@
+		*ppInterface = static_cast<IUIAutomationActiveTextPositionChangedEventHandler*>(this);
+		AddRef();
+		return S_OK;
+	}
+	*ppInterface = nullptr;
+	return E_NOINTERFACE;
+}
+
+// IUIAutomationEventHandler method
+HRESULT STDMETHODCALLTYPE RateLimitedEventHandler::HandleAutomationEvent(IUIAutomationElement* sender, EVENTID eventID) {
+	LOG_DEBUG(L"RateLimitedUIAEventHandler::HandleAutomationEvent called");
+	LOG_DEBUG(L"Queuing automationEvent for eventID "<<eventID);
+	return queueEvent<AutomationEventRecord_t>(sender, eventID);
+}
+
+// IUIAutomationFocusEventHandler method
+HRESULT STDMETHODCALLTYPE RateLimitedEventHandler::HandleFocusChangedEvent(IUIAutomationElement* sender) {
+	LOG_DEBUG(L"RateLimitedUIAEventHandler::HandleFocusChangedEvent called");
+	return queueEvent<FocusChangedEventRecord_t>(sender);
+}
+
+// IUIAutomationPropertyChangedEventHandler method
+HRESULT STDMETHODCALLTYPE RateLimitedEventHandler::HandlePropertyChangedEvent(IUIAutomationElement* sender, PROPERTYID propertyID, VARIANT newValue) {
+	LOG_DEBUG(L"RateLimitedUIAEventHandler::HandlePropertyChangedEvent called");
+	return queueEvent<PropertyChangedEventRecord_t>(sender, propertyID, newValue);
+}
+
+// IUIAutomationNotificationEventHandler method
+HRESULT STDMETHODCALLTYPE RateLimitedEventHandler::HandleNotificationEvent(IUIAutomationElement* sender, NotificationKind notificationKind, NotificationProcessing notificationProcessing, BSTR displayString, BSTR activityID) {
+	LOG_DEBUG(L"RateLimitedUIAEventHandler::HandleNotificationEvent called");
+	return queueEvent<NotificationEventRecord_t>(sender, notificationKind, notificationProcessing, displayString, activityID);
+}
+
+// IUIAutomationActiveTextPositionchangedEventHandler method
+HRESULT STDMETHODCALLTYPE RateLimitedEventHandler::HandleActiveTextPositionChangedEvent(IUIAutomationElement* sender, IUIAutomationTextRange* range) {
+	LOG_DEBUG(L"RateLimitedUIAEventHandler::HandleActiveTextPositionChangedEvent called");
+	return queueEvent<ActiveTextPositionChangedEventRecord_t>(sender, range);
+}
+
+void RateLimitedEventHandler::flush() {
+	LOG_DEBUG(L"RateLimitedUIAEventHandler::flush called");
+	decltype(m_eventRecords) eventRecordsCopy;
+	decltype(m_eventRecordsByKey) eventRecordsByKeyCopy;
+	{ std::lock_guard lock(mtx);
+		eventRecordsCopy.swap(m_eventRecords);
+		eventRecordsByKeyCopy.swap(m_eventRecordsByKey);
+	}
+
+	// Emit events
+	LOG_DEBUG(L"RateLimitedUIAEventHandler::flush: Emitting events...");
+	for(const auto& recordVar: eventRecordsCopy) {
+		std::visit([this](const auto& record) {
+			this->emitEvent(record);
+		}, recordVar);
+	}
+	/*
+	unsigned int count = std::accumulate(eventRecordsByKeyCopy.begin(), eventRecordsByKeyCopy.end(), 0, [](const auto& acc, const auto& i) {
+		auto count = i.second.second;
+		if(count > 1) {
+			return acc + count;
+		}
+		return acc;
+	});
+	if(count > 0) {
+		Beep(440 + (count*10), 40);
+	}
+	*/
```

Comment:
I assume this was for debugging to figure out how many events were coalesced? I guess this might be useful in future, but if you're going to keep this, please add a comment above it explaining what it does and what it's useful for.